vnext/Microsoft.ReactNative/Fabric/Composition/Modal/WindowsModalHostViewComponentView.cpp

```diff
@@ -174,20 +218,42 @@ void WindowsModalHostComponentView::RegisterWndClass() noexcept {
   registered = true;
 }
 
+// childComponentView - reference to the child component view
+// index - the position in which the childComponentView should be mounted
 void WindowsModalHostComponentView::MountChildComponentView(
     const winrt::Microsoft::ReactNative::ComponentView &childComponentView,
     uint32_t index) noexcept {
-  // Disabled due to partial Modal implementation. Tracking re-enablement with task list here:
-  // https://github.com/microsoft/react-native-windows/issues/11157 assert(false);
-  base_type::MountChildComponentView(childComponentView, index);
+  EnsureModalCreated();
+  // Skip calling ViewComponentView::MountChildComponentView since we override adding it to the rootVisual below
+  ComponentView::MountChildComponentView(childComponentView, index);
+
+  // Handle index offset and ensure visual
+  indexOffsetForBorder(index);
+  ensureVisual();
+
+  if (auto compositionChild = childComponentView.try_as<ComponentView>()) {
+    auto containerChildren = m_rootVisual.Children();
+    auto compVisual =
+        winrt::Microsoft::ReactNative::Composition::Experimental::MicrosoftCompositionContextHelper::InnerVisual(
+            compositionChild->OuterVisual());
+
+    if (index == 0) {
+      containerChildren.InsertAtBottom(compVisual);
+    } else {
+      auto insertAfter = containerChildren.First();
+      for (uint32_t i = 1; i < index; i++) {
+        insertAfter.MoveNext();
+      }
+      containerChildren.InsertAbove(compVisual, insertAfter.Current());
+    }
+  }
 }
 
 void WindowsModalHostComponentView::UnmountChildComponentView(
     const winrt::Microsoft::ReactNative::ComponentView &childComponentView,
     uint32_t index) noexcept {
-  // Disabled due to partial Modal implementation.Tracking re-enablement with task list here : https : //
-  // github.com/microsoft/react-native-windows/issues/11157 assert(false);
-  base_type::UnmountChildComponentView(childComponentView, index);
+  ComponentView::UnmountChildComponentView(childComponentView, index);
+  HideOnUIThread();
```

Comment:
I looked into this and it turns out FabricUIManager calls Create and Delete on the Modal when visibility changes. This works fine for creation because it triggers updateProps, which then creates the window. But on deletion, it only removes the Modal from the tree, where we actually need it to run the code that deletes the window too.

What I ended up doing was adding a check when deleting, so if it's a Modal, it runs the code to delete the window. I'm a little unsure about this solution though, since it relies on string comparisons, and those can be kinda expensive performance-wise.